### PR TITLE
Refactor API codebase

### PR DESCRIPTION
**Pull request by FedericoCeratto**
This is a refactor for the API code. The HTTP API itself is not breaking backward compatibility.

- [x] Refactor API codebase in a dedicated directory
- [x] Consolidate Probe Services
  - [ ] Authentication
  - [x] Collector
  - [x] Dynamic URL test list with priorities
- [x] Implement new canning-equivalent 
- [x] Remove dependencies from batch pipeline
  - [x] Measurement table
  - [x] Reimplement private api to move away from ooexpl_* tables
- [x] Implement packaging to support CI deployment
- [x] Run as a daemon using a systemd unit file
- [x] Cleanup dependencies and use stable libraries from Buster
- [x] Generate API docs/swagger automatically with flasgger

Running at: https://ams-pg.ooni.org/apidocs/ and used by https://explorer-test.ooni.io/

### CODE REVIEW

File: newapi/debian/etc/nginx/sites-available/ooni-api.conf [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# deployed by ooni-api .deb package
+log_format ooniapi_fmt '"$request" $status';
+server {
```
Comment (bassosimone) [inline newapi/debian/etc/nginx/sites-available/ooni-api.conf:3]: Does this expose the API on port `80`? If so, do we have a use case for API over cleartext HTTP?

File: newapi/debian/etc/nginx/sites-available/ooni-api.conf [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  server_name _;
+  access_log syslog:server=unix:/dev/log,tag=ooniapi,severity=info ooniapi_fmt;
+  location / {
+    proxy_pass http://127.0.0.1:8000;
+    proxy_set_header Host $host;
+    proxy_set_header X-Real-IP $remote_addr;
+  }
+}
+
+#FIXME
```
Comment (bassosimone) [inline newapi/debian/etc/nginx/sites-available/ooni-api.conf:15]: What does this `FIXME` refer to?

**Review by bassosimone — COMMENTED** [review]
Have read roughly 1/4 of the diff so far and flagged a bunch of places of interest. 🥫

**Comment by FedericoCeratto** [comment]
Required for https://github.com/ooni/pipeline/pull/324

**Pull request merged**